core/templates/dev/head/pages/exploration_player/ExtractImageFilenamesFromStateService.js

```diff
@@ -108,30 +108,108 @@ oppia.factory('ExtractImageFilenamesFromStateService', [
+        //    height: heightOfImage
+        // }
+        // So, we create an object similar to it for the images which had
+        // filenames as the attribute's value with default dimensions of
+        // 500px x 200px
+        if (!filepathObject.width) {
+          var filename = filepathObject;
+          filepathObject = {
+            name: filename,
+            width: 500,
+            height: 200
+          };
+        }
+        fileInfo.push(filepathObject);
       }
-      // The name in the array is stored as '"image.png"'. We need to remove
-      // the inverted commas. We remove the first and the last character from
-      // the string (name).
-      filenames = filenames.map(function(filename) {
-        return filename.slice(1, filename.length - 1);
+      return fileInfo;
+    };
+
+    /**
+    * Gets the filenames of the images from the html provided.
+    * @param {string} htmlStr - The string from which the filenames of the
+    *                           images should be extracted.
+    */
+    var _getImageFilenamesFromFilepathValue = function(htmlStr) {
+      var fileInfos = (
+        _extractFilepathValueFromOppiaNonInteractiveImageTag(htmlStr));
+      return fileInfos.map(function(fileInfo) {
+        return fileInfo.name;
+      });
+    };
+
+    /**
+    * Gets the dimensions of the images from the html provided.
+    * @param {string} htmlStr - The string from which the dimensions of the
+    *                           images should be extracted.
+    */
+    var _getImageDimensionsFromFilepathValue = function(htmlStr) {
+      var fileInfos = (
+        _extractFilepathValueFromOppiaNonInteractiveImageTag(htmlStr));
+      var fileDimensions = {};
+      fileInfos.forEach(function(fileInfo){
+        var filename = fileInfo.name;
+        fileDimensions[filename] = {
+          width: fileInfo.width,
+          height: fileInfo.height
+        };
+      });
+      return fileDimensions;
+    };
+
+    /**
+    * Gets the dimensions of the images from the state provided.
+    * @param {object} state - The state from which the dimensions of the
+    *                           images should be extracted.
+    */
+    var _getImageDimensionsInState = function(state) {
+      var fileDimensions = {};
+      if (state.interaction.id === INTERACTION_TYPE_IMAGE_CLICK_INPUT) {
+        var filepathObject = (
+          state.interaction.customizationArgs.imageAndRegions.value.imagePath);
+        if (!filepathObject.width) {
+          var filename = filepathObject;
+          filepathObject = {
+            name: filename,
+            width: 500,
+            height: 200
+          };
+        }
+        fileDimensions[filepathObject.name] = {
+          width: filepathObject.width,
+          height: filepathObject.height
+        };
+      }
+      var allHtmlOfState = _getAllHtmlOfState(state);
+      allHtmlOfState.forEach(function(htmlStr) {
+        fileDimensions = Object.assign(fileDimensions,
```

Comment:
I think you don't need `fileDimensions = ` because `Object.assign` will modeify the `fileDimensions` anyways.

Reply:
Done!